x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java

```diff
@@ -329,12 +318,17 @@ public static MlConfigVersion fromString(String str) {
         if (str.startsWith("8.10.") || str.equals("8.11.0")) {
             return V_10;
         }
-        Matcher matcher = Pattern.compile("^(\\d+)\\.0\\.0$").matcher(str);
-        int versionNum;
-        if (matcher.matches() == false || (versionNum = Integer.parseInt(matcher.group(1))) < 10) {
-            return fromVersion(Version.fromString(str));
+        Matcher matcher = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)(?:-\\w+)?$").matcher(str);
```

Comment:
I might have missed this from an earlier review, but why does ML config version look like a semantic version? Could it just be an integer going forward, like other version concepts we have now (transport version, index version, etc)?

Reply:
Old nodes in mixed version clusters need to be able to parse the text representation of the newer versions without throwing exceptions.